iris_grib/tests/unit/grib_phenom_translation/test_timecodes.py

```diff
@@ -0,0 +1,274 @@
+        self._init_minimal_message(edition=edition)
+        # Also set a time-code, if given.
+        if time_code is not None:
+            self.set_timeunit_code(time_code)
+        # Finally, add any remaining passed key-values.
+        self.update(**kwargs)
+
+    def _init_minimal_message(self, edition=1):
+        # Set values for all the required keys.
+        # 'edition' controls the edition-specific keys.
+        self.update({
+            'Ni': 1,
+            'Nj': 1,
+            'numberOfValues': 1,
+            'alternativeRowScanning': 0,
+            'centre': 'ecmf',
+            'year': 2007,
+            'month': 3,
+            'day': 23,
+            'hour': 12,
+            'minute': 0,
+            'indicatorOfUnitOfTimeRange': 1,
+            'shapeOfTheEarth': 6,
+            'gridType': 'rotated_ll',
+            'angleOfRotation': 0.0,
+            'iDirectionIncrementInDegrees': 0.036,
+            'jDirectionIncrementInDegrees': 0.036,
+            'iScansNegatively': 0,
+            'jScansPositively': 1,
+            'longitudeOfFirstGridPointInDegrees': -5.70,
+            'latitudeOfFirstGridPointInDegrees': -4.452,
+            'jPointsAreConsecutive': 0,
+            'values': np.array([[1.0]]),
+            'indicatorOfParameter': 9999,
+            'parameterNumber': 9999,
+        })
+        # Add edition-dependent settings.
+        self['edition'] = edition
+        if edition == 1:
+            self.update({
+                'startStep': 24,
+                'timeRangeIndicator': 1,
+                'P1': 2, 'P2': 0,
+                # time unit - needed AS WELL as 'indicatorOfUnitOfTimeRange'
+                'unitOfTime': 1,
+                'table2Version': 9999,
+            })
+        if edition == 2:
+            self.update({
+                'iDirectionIncrementGiven': 1,
+                'jDirectionIncrementGiven': 1,
+                'uvRelativeToGrid': 0,
+                'forecastTime': 24,
+                'productDefinitionTemplateNumber': 0,
+                'stepRange': 24,
+                'discipline': 9999,
+                'parameterCategory': 9999,
+                'tablesVersion': 4
+            })
+
+    def set_timeunit_code(self, timecode):
+        # Do timecode setting (somewhat edition-dependent).
+        self['indicatorOfUnitOfTimeRange'] = timecode
+        if self['edition'] == 1:
+            # for some odd reason, GRIB1 code uses *both* of these
+            # NOTE kludge -- the 2 keys are really the same thing
+            self['unitOfTime'] = timecode
+
+
+class TestGribTimecodes(tests.IrisGribTest):
+    def _run_timetests(self, test_set):
+        # Check the unit-handling for given units-codes and editions.
+
+        # Operates on lists of cases for various time-units and grib-editions.
+        # Format: (edition, code, expected-exception,
+        #          equivalent-seconds, description-string)
+        for test_controls in test_set:
```

Comment:
Just above this line, this iris test had this: https://github.com/SciTools/iris/blob/master/lib/iris/tests/test_grib_load_translations.py#L206

That allowed the test to work because it avoided the gribapi type-check.  Without that line, gribapi messes everything up.